### PR TITLE
Adds new arsonist quest and unlocks her as a merchant after completion.

### PR DESCRIPTION
**Pull request by Brian-Otten**
#### Summary
```SUMMARY: Content "Adds new quest and stock to the arsonist"```

#### Purpose of change
The arsonist currently says she will sell you some molotovs but doesn't really, and she is flagged as a potential quest giver in the json files.

#### Describe the solution
The arsonist now has a quest for the player to bring her 2 bags of commercial fertilizer so she can expand her arsenal with bombs rather than just molotovs. If you complete the quest, she will give you 10 molotovs as a reward, and also become available as a trader. 
She has a stock of explosives (molotovs and pipebombs), some fuses, and a chance to sell the anarchist book or some spare commercial fertilizer.

#### Testing
Load up the game.
Do the quest successfully, wandering through all the dialog options to check for bugs.
Check that trade was only unlocked after quest completion.
Trade with npc.

#### Additional context
The way the game loads stock for merchants seems to be a bit of a mess right now so the stock available on this (and all other merchants) is quite random. I hope to circle around and try to tackle that in depth later.

image

### CODE REVIEW

File: data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json [hunk 1]
```diff
@@ -79,12 +109,47 @@
...
+    "description": "Bring the arsonist 42 kg of commercial fertilizer",
+    "difficulty": 1,
+    "value": 500,
+    "origins": [ "ORIGIN_SECONDARY" ],
+    "goal": "MGOAL_FIND_ITEM",
+    "item": "fertilizer_commercial",
+    "count": 120,
+    "dialogue": {
+      "describe": "I could use some help getting some materials.",
+      "offer": "I could use some help getting some materials.  Molotovs are nice for burning down buildings but aren't all that for keeping yourself safe.  If you could get me two bags of commercial fertilizer i could make something a lot more potent.",
```
Comment (Night-Pryanik) [inline data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json:131]:
```suggestion
      "offer": "I could use some help getting some materials.  Molotovs are nice for burning down buildings but aren't all that for keeping yourself safe.  If you could get me two bags of commercial fertilizer I could make something a lot more potent.",
```

File: data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json [hunk 2]
```diff
@@ -79,12 +109,47 @@
...
+      "accepted": "Oh man, thanks so much friend.  You won't regret it.",
+      "rejected": "Think it over.",
+      "advice": "Commercial fertilizer tends to be made from ammonium nitrate, which is also usable as an explosive.  Try looking for some around the abandoned farms",
```
Comment (Night-Pryanik) [inline data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json:134]:
```suggestion
      "advice": "Commercial fertilizer tends to be made from ammonium nitrate, which is also usable as an explosive.  Try looking for some around the abandoned farms.",
```

File: data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json [hunk 3]
```diff
@@ -79,12 +109,47 @@
...
+      "inquire": "Did you find any fertilizer?  Are you sure this is the ammonium nitrate kind?",
+      "success": "This is great! I'll get to work making something of this.  You can have some of my molotovs, i won't need quite so many now.",
```
Comment (Night-Pryanik) [inline data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json:136]:
```suggestion
      "success": "This is great!  I'll get to work making something of this.  You can have some of my Molotovs, I won't need quite so many now.",
```

File: data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json [hunk 4]
```diff
@@ -79,12 +109,47 @@
...
+      "success_lie": "I guess i could convince the shelter to sell me some.",
```
Comment (Night-Pryanik) [inline data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json:137]:
```suggestion
      "success_lie": "I guess I could convince the shelter to sell me some.",
```

File: data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json [hunk 5]
```diff
@@ -79,12 +109,47 @@
...
+      "failure": "Damned.  Perhaps i wasn't the only one that thought of this."
```
Comment (Night-Pryanik) [inline data/json/npcs/refugee_center/surface_visitors/NPC_arsonist.json:138]:
```suggestion
      "failure": "Damn.  Perhaps I wasn't the only one who thought of this."
```

**Comment by Brian-Otten** [comment]
Thanks for the second pair of eyes, changes committed.

**Comment by AlexMooney** [comment]
Could you please also add rebar to the arsonist stocks?  Probably nails too, since you get those in great quantities from burning down buildings.

**Comment by Brian-Otten** [comment]
Added rebar and nails and checked they were working properly. good idea.

**Pull request merged**